core/shared/src/main/scala/monocle/function/Plated.scala

```diff
@@ -62,4 +65,19 @@ trait PlatedFunctions {
   def transformOf[A](l: Setter[A, A])(f: A => A)(a: A): A =
     l.modify(b => transformOf(l)(f)(f(b)))(a)
 
-}
+  /** transforming counting changes */
+  def transformC[A: Plated](pf: PartialFunction[A, A])(a: A): (Int, A) = {
```

Comment:
maybe we could make the name more explicit since this method is probably not going to be called frequently. What about `transformCounting`?